src/ibusunicode.c

```diff
@@ -424,7 +424,10 @@ ibus_unicode_data_list_deserialize (GVariant *variant,
         IBusUnicodeData *data =
                 IBUS_UNICODE_DATA (ibus_serializable_deserialize (
                         unicode_variant));
+        if(data){
+        g_object_ref(data);  // Ref the object when adding to list
```

Comment:
It would be good to revert your change.
I don't think `g_object_rer()` is needed here. If you put it here, `list->data->ref_count` is still 1 after you call ` ibus_destroy_list_and_data`.